
# Multiple Linear Regression – Corrected Statistical Evaluation

## Project Overview
This notebook evaluates the relationship between multiple marketing channels and Sales using Multiple Linear Regression.


In [5]:

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm
from statsmodels.stats.outliers_influence import variance_inflation_factor
from statsmodels.stats.diagnostic import het_breuschpagan


In [6]:

# Load dataset
df = pd.read_csv('marketing_sales_data.csv')

# Rename column
df = df.rename(columns={'Social Media': 'Social_Media'})

# Convert categorical TV variable into dummy variables
df = pd.get_dummies(df, columns=['TV'], drop_first=True)

df.head()


,Radio,Social_Media,Influencer,Sales,TV_Low,TV_Medium
0,3.518070,2.293790,Micro,55.261284,True,False
1,7.756876,2.572287,Mega,67.574904,True,False
2,20.348988,1.227180,Micro,272.250108,False,False
3,20.108487,2.728374,Mega,195.102176,False,True
4,31.653200,7.776978,Nano,273.960377,False,False


In [7]:

# Define predictors and target variable
X = df[['Radio', 'Social_Media', 'TV_Low', 'TV_Medium']]
y = df['Sales']

# Convert to numeric
X = X.astype(float)
y = y.astype(float)

# Add constant
X = sm.add_constant(X)

# Fit regression model
model = sm.OLS(y, X).fit()

# Display model summary
print(model.summary())


                            OLS Regression Results                            
Dep. Variable:                  Sales   R-squared:                       0.904
Model:                            OLS   Adj. R-squared:                  0.903
Method:                 Least Squares   F-statistic:                     1335.
Date:                Mon, 08 Jun 2026   Prob (F-statistic):          7.29e-287
Time:                        12:12:31   Log-Likelihood:                -2713.9
No. Observations:                 572   AIC:                             5438.
Df Residuals:                     567   BIC:                             5460.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
const          218.6473      6.290     34.761   

In [8]:

# Variance Inflation Factor (VIF)
vif_data = pd.DataFrame()
vif_data['Variable'] = X.columns
vif_data['VIF'] = [
    variance_inflation_factor(X.values, i)
    for i in range(X.shape[1])
]

print(vif_data)


       Variable        VIF
0         const  28.986454
1         Radio   3.462738
2  Social_Media   1.658537
3        TV_Low   4.064612
4     TV_Medium   2.217533



# Statistical Evaluation and Interpretation

## Regression Equation

Sales = 218.6473 + (2.9891 × Radio) + (-0.15 × Social_Media) + (-154.3121 × TV_Low) + (-75.3279 × TV_Medium)

## Adjusted R-squared

The model achieved an Adjusted R-squared value of **0.9033**.

This means that approximately **90.33%** of the variation in Sales is explained by the predictor variables in the model.

## F-statistic

The regression model produced an F-statistic of **1335.0705** with a p-value of **0.0**.

Since the F-statistic p-value is below 0.05, the overall regression model is statistically significant.

## Coefficient Interpretation

### Radio
- Coefficient: **2.9891**
- p-value: **0.0**

Holding other variables constant, a one-unit increase in Radio advertising spend is associated with an increase of **2.9891** units in Sales.

### Social Media
- Coefficient: **-0.15**
- p-value: **0.823701**

Holding other variables constant, a one-unit increase in Social Media advertising spend is associated with an increase of **-0.15** units in Sales.

### TV_Low
- Coefficient: **-154.3121**
- p-value: **0.0**

Compared to the reference TV category, the TV_Low category changes Sales by **-154.3121** units on average.

### TV_Medium
- Coefficient: **-75.3279**
- p-value: **0.0**

Compared to the reference TV category, the TV_Medium category changes Sales by **-75.3279** units on average.

## Multicollinearity

The VIF values indicate that multicollinearity is not severe because the values are relatively low.

## Final Recommendation

Based on the regression results:
- Variables with statistically significant p-values should receive greater marketing investment.
- Radio and Social Media advertising positively influence Sales.
- The organization should prioritize channels with the strongest significant coefficients to maximize ROI.
